File: .scripts/setAllMessageSilent/setAllMessageSilent.py
import json
import os
import sys

def setSilentTrue(obj):
    if isinstance(obj, list):
        for item in obj:
            setSilentTrue(item)
    elif isinstance(obj, dict):
        if obj.get("name") == "Send Message":
            if "data" not in obj or not isinstance(obj["data"], dict):
                obj["data"] = {}
            obj["data"]["silent"] = True
        for value in obj.values():
            setSilentTrue(value)

# Load the JSON file
if len(sys.argv) > 1:
    filePath = sys.argv[1].strip('"').strip()
else:
    filePath = input("📄 Drag your 'data.json' file here: ").strip('"').strip()
if os.path.basename(filePath) != "data.json":
    print("❌ The file must be named exactly 'data.json'.")
    exit(1)

with open(filePath, "r", encoding="utf-8") as f:
    data = json.load(f)

# Modify all Send Message actions
setSilentTrue(data)

# Save the modified JSON back to file
outputPath = os.path.join(os.path.dirname(filePath), "modified_data.json")
with open(outputPath, "w", encoding="utf-8") as f:
    json.dump(data, f, indent=2)

File: .scripts/setAllMessageSilent/setAllMessageSilent.js
const fs = require("node:fs")
const path = require("node:path")
const readline = require("node:readline")

function setSilentTrue(obj) {
  if (Array.isArray(obj)) {
    obj.forEach(setSilentTrue)
  } else if (typeof obj === "object" && obj !== null) {
    if (obj.file === "sendmessage.js") {
      if (!obj.data) obj.data = {}
      obj.data.silent = true
    }
    for (const key in obj) {
      setSilentTrue(obj[key])
    }
  }
}

function modifyFile(filePath) {
  if (path.basename(filePath) !== "data.json") {
    console.error(`File Isn't "data.json"!`)
    return
  }

  try {
    const rawData = fs.readFileSync(path.normalize(filePath), "utf-8")
    const data = JSON.parse(rawData)

    setSilentTrue(data)

    const outputFilePath = path.join(
      path.dirname(filePath),
      "modified_" + path.basename(filePath)
    )

    fs.writeFileSync(outputFilePath, JSON.stringify(data, null, 2), "utf-8")
    console.log(`Modified File Saved As: ${outputFilePath}`)
  } catch (err) {
    console.error("Failed To Process The File:", err.message)
  }
}

// Always prompt user for the file path
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

rl.question(`Drag the "data.json" file into this terminal: `, (answer) => {
  const filePath = answer.trim().replaceAll(`"`, ``)
  modifyFile(filePath)
  rl.close()
})